### PR TITLE
feat: take source and target epsg, if not the same transform TDE-699

### PR DESCRIPTION
**Pull request by MDavidson17**

### CODE REVIEW

**Review by paulfouquet — APPROVED** [review]

File: scripts/gdal/gdal_preset.py [hunk 1]
```diff
@@ -76,12 +73,12 @@
...
 
 
-def get_gdal_command(preset: str, convert_from: Optional[str] = None) -> List[str]:
+def get_gdal_command(preset: str, source_epsg: str, convert_from: Optional[str] = None) -> List[str]:
     get_log().info("gdal_preset", preset=preset)
     gdal_command: List[str] = ["gdal_translate"]
 
     gdal_command.extend(BASE_COG)
-    gdal_command.extend(NZTM_SOURCE)
+    gdal_command.extend(["-a_srs", f"EPSG:{source_epsg}"])
```
Comment (paulfouquet) [inline scripts/gdal/gdal_preset.py:81]: Can you put back the comment `# Force the source projection to source_epsg`

**Review by amfage — APPROVED** [review]

File: scripts/gdal/gdal_preset.py [hunk 2]
```diff
@@ -132,3 +130,10 @@ def get_alpha_command() -> List[str]:
         # Ensure the target has a alpha channel
         "-dstalpha",
     ]
+
+
+def get_transform_srs_command(target_epsg: str) -> List[str]:
+    """
+    Get a "Gdalwarp" command to transform the srs
+    """
+    return ["gdalwarp", "-t_srs", f"EPSG:{target_epsg}"]
```
Comment (blacha) [inline scripts/gdal/gdal_preset.py:139]: What resampling method does `gdalwarp` run by default? I believe we need to use `bilinear` (?)

Comment (MDavidson17) [reply scripts/gdal/gdal_preset.py:139]: yes I was just thinking this, shall we match Rebecca's gist:
`gdalwarp -t_srs 'EPSG:2193' -r bilinear -tr 10 -10 -co compress=lzw ./output/overall_wgs84.tif ./output/overall_NZTM.tif`

Comment (blacha) [reply scripts/gdal/gdal_preset.py:139]: this is also missing the rest of the arguments, like `ALL_CPUS` that I shared earlier.

**Review by blacha — APPROVED** [review]

**Pull request merged**